blockchain/state_transition.go

```diff
@@ -234,6 +236,15 @@ func (st *StateTransition) buyGas() error {
 	// st.gasPrice : BaseFee after magma hardfork
 	mgval := new(big.Int).Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
 
+	if st.evm.ChainConfig().Rules(st.evm.Context.BlockNumber).IsOsaka {
+		if blobGas := st.blobGasUsed(); blobGas > 0 {
+			// Pay for blobGasUsed * actual blob fee
+			blobFee := new(big.Int).SetUint64(blobGas)
+			blobFee.Mul(blobFee, st.evm.Context.BlobBaseFee)
```

Comment:
Isn't it blobGasUsed * tx.blobFeeCap??

Reply:
Yes, it isn't. The actual gas deducted is blobGas*blobBaseFee.
But this doesn't meet [Block validation-2](https://github.com/kaiachain/kips/blob/fac337aad17db40ed2a6c988e03ad6b2ec9771f2/KIPs/kip-279.md#block-
validation). This can be done in a miner.

Reply:
I've checked and decided to add `balanceCheck`.
It was added at [a22c69b](https://github.com/kaiachain/kaia/pull/675/commits/a22c69b49fa9438f76fb2b0a4d01ce4c09fc51d7). Thanks.